src/core/App.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @author Lewy Blue / https://github.com/looeee
+ *
+ * parameters = {
+ *
+ *  canvas: <canvas> DOM element,
+ *
+ *  scene: new THREE.Scene(),
+ *
+ *  camera: new THREE.Camera( ... )
+ *
+ *  renderer: new THREE.WebGLRenderer( ... )
+ *
+ *  autoResize: <bool>
+ *
+ * }
+ */
+
+function App( parameters ) {
+
+	parameters = parameters || {};
+
+	if ( parameters.canvas !== undefined ) {
+
+		this.canvas = parameters.canvas;
+
+	} else {
+
+		this.canvas = document.body.appendChild( document.createElement( 'canvas' ) );
+		this.canvas.style.width = this.canvas.style.height = '100%';
+
+	}
+
+	this.scene = ( parameters.scene !== undefined ) ? parameters.scene : new THREE.Scene();
+
+	this.camera = ( parameters.camera !== undefined ) ? parameters.camera : new THREE.PerspectiveCamera( 75, this.canvas.clientWidth / this.canvas.clientHeight, 1, 1000 );
```

Comment:
I saw your comment above. If you follow my example, since the whole point of `THREE.App` is to manage stuff you should just called `updateCamera` or whatever code would normally update the cameras before rendering right after creating the camera. D.R.Y. and all that.